### ISSUE TITLE
navigation_tutorial tutorial3.launch crashes mapper

### ISSUE THREAD

**Issue by manuel-delverme** [skasperski/navigation_2d#19]
Running tutorial3.launch results in a crash  

```
[Mapper-4] process has died [pid 11039, exit code -11, cmd /opt/ros/jade/lib/nav2d_karto/mapper scan:=base_scan __name:=Mapper __log:=/home/noflip/.ros/log/b6b6ae70-1acb-11e6-b2f9-1c659dee480d/Mapper-4.log].
log file: /home/noflip/.ros/log/b6b6ae70-1acb-11e6-b2f9-1c659dee480d/Mapper-4*.log
```

position of the robot:
![screenshot](https://cloud.githubusercontent.com/assets/1309922/15276080/75c29dcc-1add-11e6-99de-fbf1f1b37489.png)

![screenshot](https://cloud.githubusercontent.com/assets/1309922/15276113/8b4e89e8-1ade-11e6-9111-fc383ea80f82.png)
(i wouldn't trust rviz here, i think it didn't refresh the position since i was in another worskpace)

```
[terminal:1]
➜  launch git:(master) ✗ roslaunch nav2d_tutorials tutorial3.launch

[terminal:2]
➜  launch git:(master) ✗ rosservice call /StartMapping 3
response: 0
➜  launch git:(master) ✗ rosservice call /StartExploration 2
response: 0

[terminal:1]
[Mapper-4] process has died [pid 13336, exit code -11, cmd /opt/ros/jade/lib/nav2d_karto/mapper scan:=base_scan __name:=Mapper __log:=/home/$USER/.ros/log/da07a914-1acc-11e6-b2f9-1c659dee480d/Mapper-4.log].
log file: /home/$USER/.ros/log/da07a914-1acc-11e6-b2f9-1c659dee480d/Mapper-4*.log

```

log file is empty

same exception happens, in another location after adding respawn="true" to mapper in the launch file

**Comment by manuel-delverme** [comment]
![screenshot](https://cloud.githubusercontent.com/assets/1309922/15276140/9ab55bcc-1adf-11e6-9903-08c64af2f2c8.png)
same thing happened here, maybe it's not the specific position

**Comment by skasperski** [comment]
Which version of ROS are you using? Have you installed binaries or build it from source? Does the crash happen immediately after the service call? 

Also check rqt_console for any hints just before the crash happens.

**Comment by manuel-delverme** [comment]
Jade binaries from Ubuntu repository
The crash happens in choke points like the ones in the screenshots 
I'll try to post rqt console but I'll take me a while

**Comment by skasperski** [comment]
Sorry for late reply. I have looked into this a bit and it seems to happen sometimes when a new map is requested. (New requests are likely to happen at narrow points.) It seems to be caused by some (changed?) behavior of libTBB inside of OpenKarto, but can't say anything for sure yet. It happens on a source-build with ros-kinetic as well.

**Comment by skasperski** [comment]
Solved in version 0.3.1.
